### PR TITLE
Simplify core elements initialization with Spring

### PR DESCRIPTION
**Pull request by jgiovaresco**
## Description

The PR tries to simplify the spring configuration related to core elements.

As we don't want to reference Spring in the Core package, declaring beans was cumbersome because we had to add them in a Configuration file manually.

This PR suggests 2 remediations based on `@ComponentScan` annotation:
- In this [commit](https://github.com/gravitee-io/gravitee-api-management/pull/6882/commits/c941303d5c5e40c307f4f28bfbde3ce9d0bfad5b) beans are discover using a regular expression. In is simple but not really readable and I'm not sure how it will behave if we create subpackages in `domain_service`...
- The alternative is to rely on custom Annotation. (check relevant commit). I've introduced 2 annotations, `@UseCase` and `@DomainService`.  Then, we can configure Component Scan to search for that annotation to discover beans.

I like the second approach for 2 reasons:
- no regular expression: the component scan is easy to read
- these annotations allow us to write inline documentation on the software architecture we'd like for APIM (a trick I read in [Living Documentation](https://www.amazon.fr/Living-Documentation-Cyrille-Martraire/dp/0134689321) book)

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-lxobdxixvj.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/ImportCRDUseCase.java [hunk 1]
```diff
@@ -48,6 +49,7 @@
  * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
  * @author GraviteeSource Team
  */
+@TransactionalUseCase
```
Comment (ytvnr) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/ImportCRDUseCase.java]: What about a `@UseCase(transactional = true)` ?

**Review by ytvnr — COMMENTED** [review]
The functional annotation `@UseCase` is definitely a better choice, but I would not have a separate `@TransactionalUseCase`

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/ImportCRDUseCase.java (hunk 1 above)
Comment (jgiovaresco) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/ImportCRDUseCase.java]: I think I will remove the `@TransactionalUseCase` and only keep `@UseCase`. The former annotation were introduce to "enable" Spring transaction. I suppose we can apply `@Transactional` annotation on all UseCase

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/UseCase.java [hunk 2]
```diff
@@ -34,12 +36,19 @@
  *        <li>Other rules are enforced by ArchUnit {@link architecture.UseCaseRulesTest}</li>
  *      <ul>
  * </p>
+ * <p>
+ *      ⚠️ This annotation rely on Spring annotation which is an exception to the rule of not using Framework in core.
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/UseCase.java]: 🇺🇸  --> `This annotation relies on....`

**Review by jourdiw — APPROVED** [review]

**Review by Okhelifi — APPROVED** [review]
Well done 👏🏼

**Review by ytvnr — APPROVED** [review]
do not forget to squash to avoid having the commit with the ComponentScan using pattern as it is not relevant anymore

**Pull request merged**